Remove osmodel.mak in favor of ../dmd/osmodel.mak

wilzbach opened the pull request:
A small step at cleaning up the Makefiles.

Similarly to https://github.com/dlang/tools/pull/227

wilzbach commented on posix.mak:
```diff
@@ -31,8 +31,11 @@
+DMD_DIR=../dmd
 
-include osmodel.mak
+$(shell [ ! -d $(DMD_DIR) ] && git clone --depth=1 $(GIT_HOME)/dmd $(DMD_DIR))
```
@CyberShadow this was needed to pass on DAutoTest. I assumed that you clone all repos on start?
Anyhow I doubt that this hurts anyone and at best it helps a few newcomers.

CyberShadow replied:
> I assumed that you clone all repos on start?

They're always there, just cleaned/reset between runs. Not sure why you'd get an error there.

andralex replied:
Let's see why the simpler solution doesn't work, this is an unpleasant complication.

wilzbach commented:
Btw it seems as if we got four new "random" coverage lines:

![image](https://user-images.githubusercontent.com/4370550/27806869-82844c12-603d-11e7-8628-24ccbf825e1a.png)

CyberShadow commented:
At least it's in paralellism, I won't feel guilty for masking those out.

wilzbach commented:
> They're always there, just cleaned/reset between runs. Not sure why you'd get an error there.
> Let's see why the simpler solution doesn't work, this is an unpleasant complication.

Alrighty - seem to have been a transient error. Removed the line & everything* is green  :)

* except for the broken Project-Tester of course.

CyberShadow commented:
> Not sure why you'd get an error there.

On second thought, I know why.

https://github.com/CyberShadow/ae/commit/0cc44962db38dbadd194850e7d86c1928a8d72fc

CyberShadow commented:
> which as far as I can see never existed

Oops, that's not true. Still looking into it

wilzbach commented:
>  I think because of the above (now-fixed) bug we slip a build bug in, and now website auto-deploy is failing.

Sorry.

> Oops, that's not true. Still looking into it

I think I can submit a PR. Continue on your bug hunt ;-)

wilzbach commented:
> I think I can submit a PR

https://github.com/dlang/dlang.org/pull/1791
